Stellarium/stellarium
# New planet magnitudes

### Description




This branch should add a few planet magnitude improvements:
- A new paper from Mallama&Hilton (2018). 
- A proper magnitude evaluation for the Moon.
- Umbra and penumbra circles.
- Eclipse magnitude in Infostring now in percent
- Earth shadow enlargement using either traditional (2%) or Danjon method.

Fixes #574, #1350, #430.

It does not solve #1115. There is an unknown positional mismatch between numerical and graphical operations, which is a separate problem.

### Screenshots (if appropriate):

### Type of change


### How Has This Been Tested?




**Test Configuration**:
* Operating system: Windows 10 20H2
* Graphics Card: irrelevant

We can 
- observe new planet magnitudes. Esp. Mars showing albedo-dependent magnitude changes
- note the first implementation of a more realistic Lunar magnitude
- see Lunar eclipse details: a small increase of the umbra edge seems to give a better match.
- The umbra circle sizes have been derived as geometrical circle drawn at Lunar distance and using the Besselian size formulations from the Explanatory supplement. There is currently a slight visual difference between the numerical InfoString, shadow visualisation (undocumented) and my geometrical circles results which is hard to trace. Maybe we need to solve aberration and possibly other light time effects before this mismatch can be closed.


## Checklist:
